modules/private/class-jetpack-private.php

```diff
@@ -0,0 +1,349 @@
+					__( 'This setting is ignored because you <a href="%s">made your site private</a>', 'jetpack' ), // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+					esc_url( admin_url( 'admin.php?page=jetpack#/security?term=private' ) )
+				),
+				array( 'a' => array( 'href' => true ) )
+			);
+			?>
+		</div>
+		<fieldset>
+		<?php
+	}
+
+	/**
+	 * Don't let search engines index private sites
+	 * or sites not deemed publicly available, like deleted, archived, spam.
+	 *
+	 * @param string $output Robots.txt output.
+	 */
+	public static function private_robots_txt( $output ) {
+		$output  = "User-agent: *\n"; // Purposefully overriding current output; we only want these rules.
+		$output .= "Disallow: /\n";
+		return $output;
+	}
+
+	/**
+	 * Output the meta tag that tells Pinterest not to allow users to pin
+	 * content from this page.
+	 * https://support.pinterest.com/entries/21063792-what-if-i-don-t-want-images-from-my-site-to-be-pinned
+	 */
+	public static function private_no_pinning() {
+		echo '<meta name="pinterest" content="nopin" />';
+	}
+
+	/**
+	 * Prevents ajax and post requests on private blogs for users who don't have permissions
+	 */
+	public static function private_blog_prevent_requests() {
+		global $pagenow;
+
+		$is_ajax_request       = defined( 'DOING_AJAX' ) && DOING_AJAX;
+		$is_admin_post_request = ( 'admin-post.php' === $pagenow );
+
+		// Make sure we are in the right code path, if not bail now.
+		if ( ! is_admin() || ( ! $is_ajax_request && ! $is_admin_post_request ) ) {
+			return;
+		}
+
+		if ( ! self::is_private_blog_user( get_current_blog_id() ) ) {
+			wp_die( esc_html__( 'This site is private.', 'jetpack' ), 403 );
+		}
+	}
+
+	/**
+	 * Prevents ajax requests on private blogs for users who don't have permissions
+	 *
+	 * @param string    $action The Ajax nonce action.
+	 * @param false|int $result The result of the nonce check.
+	 */
+	public static function private_blog_ajax_nonce_check( $action, $result ) {
+		if ( 1 !== $result && 2 !== $result ) {
+			return;
+		}
+
+		// These two ajax actions relate to wp_ajax_wp_link_ajax() and wp_ajax_find_posts()
+		// They are needed for users with admin capabilities in wp-admin.
+		// Read more at p3btAN-o8-p2.
+		if ( 'find-posts' !== $action && 'internal-linking' !== $action ) {
+			return;
+		}
+
+		// Make sure we are in the right code path, if not bail now.
+		if ( ! is_admin() || ( ! defined( 'DOING_AJAX' ) || ! DOING_AJAX ) ) {
+			return;
+		}
+
+		if ( ! self::is_private_blog_user( get_current_blog_id() ) ) {
+			wp_die( esc_html__( 'This site is private.', 'jetpack' ), 403 );
+		}
+	}
+
+	/**
+	 * Disables WordPress Rest API for external requests
+	 */
+	public static function disable_rest_api() {
+		if ( is_user_logged_in() && self::is_private_blog_user( get_current_blog_id() ) ) {
+			return;
+		}
+
+		if ( defined( 'REST_REQUEST' ) && REST_REQUEST ) {
+			return new WP_Error( 'private_site', __( 'This site is private.', 'jetpack' ), 403 );
```

Comment:
This will result in an HTTP 500. The response code needs to be specified in the `status` key of the third parameter:

```suggestion
			return new WP_Error( 'private_site', __( 'This site is private.', 'jetpack' ), array( 'status' => 403 ) );
```

Reply:
Fixed in [11bd70d](https://github.com/Automattic/jetpack/pull/10333/commits/11bd70d0ce5980dc54f310329ee4db8946517ad8).